commit d501e759caa79579cd08e76886374ba1c348c524 (refs/changes/81/2081/7)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-06-14T00:16:10+00:00 (2 years, 4 months ago)
    
    OS-6177 avoid checking content-md5 on imgadm import when a checksum exists on image metadata

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index 101bafb0..12f0f6f9 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -6,6 +6,11 @@ Known issues:
   Docker Registry v1, which is now deprecated.
 
 
+## 3.7.2
+
+- OS-6177 avoid checking content-md5 on imgadm import when a checksum exists on
+  image metadata
+
 ## 3.7.1
 
 - OS-5823 Allow 'imgadm install' to be able to install a type=docker image.
diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index 8b733c31..64e7c750 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2015, Joyent, Inc. All rights reserved.
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
  *
  * * *
  * The main imgadm functionality. The CLI is a light wrapper around this tool.
@@ -1424,10 +1424,12 @@ IMGADM.prototype._downloadImageFile = function _downloadImageFile(opts, cb) {
             if (!zstream && opts.imgMeta.checksum) {
                 ctx.checksum = opts.imgMeta.checksum.split(':');
                 ctx.checksumHash = crypto.createHash(ctx.checksum[0]);
-            }
-            if (ctx.stream.headers['content-md5']) {
+                log.debug({uuid: uuid}, 'creating %s checksum hash',
+                    ctx.checksum[0]);
+            } else if (ctx.stream.headers['content-md5']) {
                 ctx.contentMd5 = ctx.stream.headers['content-md5'];
                 ctx.md5sumHash = crypto.createHash('md5');
+                log.debug({uuid: uuid}, 'creating content-md5 checksum hash');
             }
             ctx.stream.on('data', function (chunk) {
                 if (opts.bar)
diff --git a/src/img/package.json b/src/img/package.json
index 66b86a93..c9e0ba5b 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.7.1",
+  "version": "3.7.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
